keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1525 @@
+of kubelet that doesn't support this feature, it will ignore the pod-level resouce specifications and continue
+to use the aggregated container-level values (backward compatible). Applications must be prepared for this version
+skew implementation.
+
+For users wanting to use this feature, it is user's responsibility to use the correct version of kube-scheduler,
+kube-apiserver and kubelet otherwise the feature will not work as expected.
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: PodLevelResources
+  - Components depending on the feature gate: kubelet, kube-apiserver, kube-scheduler
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No. `resources` must be set/unset in `spec` at pod-level to enable/disable this feature.
+    The workloads need to restart to disable the feature.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? No. Only the workloads need to restart.
+
+###### Does enabling the feature change any default behavior?
+
+No. This feature is guarded by a feature gate, and requires setting Pod level `resource` stanza explicitly.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Once the feature gates are disabled, pods created with pod level resource spec need to be deleted and recreated with only container level resources.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Rollbacks will require disabling the feature, and recreating the pods that were using this new feature.
+Once the feature is reenabled, the existing workloads will run as usual since they don't have `resources` set at pod level.
+The new `resources` field will become available again for use in the Pod Spec for existing and new workloads.
+
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, the tests will be added along with alpha implementation. 
+* Validate that scheduler considers pod level resources when the feature is enabled, and falls back to container level resources 
+when the feature is disabled. 
+* Validate the kubelet is using pod level resources in cgroup settings if the feature is enabled, and uses container level values
```

Comment:
Also we need to make sure that pods using the feature can only be created when the feature gate is enabled.